machinery-helper/machinery_helper_test.go

```diff
@@ -0,0 +1,191 @@
+// modify it under the terms of version 3 of the GNU General Public License as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE about this file by physical or electronic mail,
+// you may find current contact information at www.suse.com
+
+package main
+
+import (
+  "testing"
+  "reflect"
+  "os"
+  "time"
+  "github.com/nowk/go-fakefileinfo"
+)
+
+func doTestParseRpmLine(t *testing.T, line string, expected_file_type string, expected_file_name string, expected_link_target string) {
+  file_type, file_name, link_target := parseRpmLine(line)
+
+  if file_type != expected_file_type {
+    t.Errorf("parseRpmLine('%s') file type = '%s', want '%s'", line, file_type, expected_file_type)
+  }
+  if file_name != expected_file_name {
+    t.Errorf("parseRpmLine('%s') file name = '%s', want '%s'", line, file_name, expected_file_name)
+  }
+  if link_target != expected_link_target {
+    t.Errorf("parseRpmLine('%s') file type = '%s', want '%s'", line, link_target, expected_link_target)
+  }
+}
+
+func TestParseRpmLineFile(t *testing.T) {
+  line := "-rw-r--r--    1 root    root                 18234080 Mar 31 11:40 /usr/lib64/libruby2.0-static.a"
+
+  expected_file_type := "-"
+  expected_file_name := "/usr/lib64/libruby2.0-static.a"
+  expected_link_target := ""
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestParseRpmLineDir(t *testing.T) {
+  line := "drwxr-xr-x    2 root    root                        0 Mar 31 11:45 /usr/include/ruby-2.0.0/x86_64-linux/ruby"
+
+  expected_file_type := "d"
+  expected_file_name := "/usr/include/ruby-2.0.0/x86_64-linux/ruby"
+  expected_link_target := ""
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestParseRpmLineLink(t *testing.T) {
+  line := "lrwxrwxrwx    1 root    root                       19 Mar 31 11:45 /usr/lib64/libruby2.0.so -> libruby2.0.so.2.0.0"
+
+  expected_file_type := "l"
+  expected_file_name := "/usr/lib64/libruby2.0.so"
+  expected_link_target := "libruby2.0.so.2.0.0"
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestParseRpmLineFileSpaces(t *testing.T) {
+  line := "-rw-r--r--    1 root    root                    61749 Jun 26 01:56 /usr/share/kde4/templates/kipiplugins_photolayoutseditor/data/templates/a4/h/Flipping Tux Black.ple"
+
+  expected_file_type := "-"
+  expected_file_name := "/usr/share/kde4/templates/kipiplugins_photolayoutseditor/data/templates/a4/h/Flipping Tux Black.ple"
+  expected_link_target := ""
```

Comment:
don't use underscores in Go names; var expected_link_target should be expectedLinkTarget